src/syntax.c

```diff
@@ -8765,6 +8765,8 @@ hl_combine_attr(int char_attr, int prim_attr)
 	else
 	{
 	    vim_memset(&new_en, 0, sizeof(new_en));
+	    new_en.ae_u.cterm.bg_rgb = INVALCOLOR;
+	    new_en.ae_u.cterm.fg_rgb = INVALCOLOR;
```

Comment:
Thank you for the detailed bug report and proposing the patch for that.

Looks some CI tests about the builds not supporting `termguicolors` failed.  To fix that, how about enclosing those two new lines like this?
```c
#ifdef FEAT_TERMGUICOLORS
(your new code)
 #endif
```